ckanext/data_qld/resource_visibility/helpers.py

```diff
@@ -61,56 +61,30 @@ def has_user_permission_for_org(org_id, user_obj, permission):
 def process_resources(data_dict, user_obj):
     """
     Show or hide resources based on
-    the resource_visibility value and user.
+    the resource_visible value and user.
     """
-    # Loop each resources,
-    # If resource is `Resource NOT visible/Pending acknowledgement` and
-    # if user is NOT Members, Editors, Admins or sysadmin, remove the resource.
     is_sysadmin = user_obj is not None and user_obj.sysadmin
-
     if not is_sysadmin:
+        resources = data_dict.get('resources', [])
+        # if user is NOT Member, Editor, Admin remove the resource.
         if not has_user_permission_for_org(data_dict.get('owner_org'), user_obj, 'read'):
-            resources = data_dict.get('resources', [])
-            options = get_select_field_options('resource_visibility')
-            de_identified_data = data_dict.get('de_identified_data', 'NO') == 'YES'
-
+            # Loop each resources and remove resource
+            # If resource_visible is `FALSE` or
+            # resource_visible is `TRUE` and governance_acknowledgement is `NO` and de_identified_data is `YES`
+            de_identified_data = data_dict.get('de_identified_data', 'NO')
             for resource in list(resources):
-                resource_visibility = resource.get('resource_visibility', '')
-                # Value of options[2] == Resource NOT visible/Pending acknowledgement.
-                if resource_visibility == options[2].get('value') if len(options) >= 3 else False \
-                        or len(resource_visibility) == 0 and de_identified_data:
+                resource_visible = resource.get('resource_visible', 'FALSE')
+                governance_acknowledgement = resource.get('governance_acknowledgement', 'NO')
```

Comment:
This and line 79 aren't needed if resource_visible is False.

Reply:
Updated logic to be more clear and simple